process/economics/economicsData.go

```diff
@@ -82,62 +54,25 @@ func NewEconomicsData(args ArgsNewEconomicsData) (*economicsData, error) {
 	if check.IfNil(args.TxVersionChecker) {
 		return nil, process.ErrNilTransactionVersionChecker
 	}
-
-	err := checkValues(args.Economics)
-	if err != nil {
-		return nil, err
-	}
-
-	convertedData, err := convertValues(args.Economics)
-	if err != nil {
-		return nil, err
-	}
-
 	if check.IfNil(args.EpochNotifier) {
 		return nil, process.ErrNilEpochNotifier
 	}
-
 	if check.IfNil(args.EnableEpochsHandler) {
 		return nil, process.ErrNilEnableEpochsHandler
 	}
 
-	rewardsConfigs := make([]config.EpochRewardSettings, len(args.Economics.RewardsSettings.RewardsConfigByEpoch))
-	_ = copy(rewardsConfigs, args.Economics.RewardsSettings.RewardsConfigByEpoch)
-
-	sort.Slice(rewardsConfigs, func(i, j int) bool {
-		return rewardsConfigs[i].EpochEnable < rewardsConfigs[j].EpochEnable
-	})
-
-	gasLimitSettings := make([]config.GasLimitSetting, len(args.Economics.FeeSettings.GasLimitSettings))
-	_ = copy(gasLimitSettings, args.Economics.FeeSettings.GasLimitSettings)
-
-	sort.Slice(gasLimitSettings, func(i, j int) bool {
-		return gasLimitSettings[i].EnableEpoch < gasLimitSettings[j].EnableEpoch
-	})
-
-	// validity checked in checkValues above
-	topUpGradientPoint, _ := big.NewInt(0).SetString(rewardsConfigs[0].TopUpGradientPoint, 10)
+	err := checkValues(args.Economics)
```

Comment:
rename to something like checkEconomics Values or Config ?

Reply:
renamed